backend/fms_core/resources/sample.py

```diff
@@ -161,21 +161,29 @@ def import_obj(self, obj, data, dry_run):
 
         # TODO: This should throw a nicer warning if the individual already exists
         # TODO: Warn if the individual exists but pedigree/cohort is different
-        try:
-            individual, individual_created = Individual.objects.get_or_create(
-                name=get_normalized_str(data, "Individual ID"),
-                sex=get_normalized_str(data, "Sex", default=Individual.SEX_UNKNOWN),
-                taxon=taxon,
-                **({"pedigree": pedigree} if pedigree else {}),
-                **({"cohort": cohort} if cohort else {}),
-                **({"mother": mother} if mother else {}),
-                **({"father": father} if father else {}),
-            )
-            obj.individual = individual
-        except Exception as e:
-            individual_created = False
-            individual = None
-            errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
+
+        # If there's the individual's complete information then get/create
+        individual_is_complete = True if data["Individual ID"] and data["Sex"] and taxon else False
+        individual = None
+        if individual_is_complete:
+            try:
+                individual, individual_created = Individual.objects.get_or_create(
+                    name=get_normalized_str(data, "Individual ID"),
+                    sex=get_normalized_str(data, "Sex", default=Individual.SEX_UNKNOWN),
+                    taxon=taxon,
+                    **({"pedigree": pedigree} if pedigree else {}),
+                    **({"cohort": cohort} if cohort else {}),
+                    **({"mother": mother} if mother else {}),
+                    **({"father": father} if father else {}),
+                )
+                obj.individual = individual
+            except Exception as e:
+                individual_created = False
+                errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
+        elif data["Individual ID"] or data["Sex"] or taxon and not individual_is_complete:
```

Comment:
You should include the other individual fields for this one test (cohort, pedigree, mother, father) if for some reason those fields are set, this condition applies even if those fields are not required (not part of individual_is_complete).